src/chunk_append/chunk_append.c

```diff
@@ -281,3 +291,63 @@ contain_param_exec_walker(Node *node, void *context)
 
 	return expression_tree_walker(node, contain_param_exec_walker, context);
 }
+
+typedef struct
+{
+	Var *sort_var;
+	Var *ht_var;
+	Index sort_relid;
+	Index ht_relid;
+	Oid eq_opr;
+} find_equality_join_var_context;
+
+static bool
+find_equality_join_var_walker(Node *node, find_equality_join_var_context *context)
+{
+	if (node == NULL)
+		return false;
+
+	if (IsA(node, OpExpr))
+	{
+		OpExpr *op = castNode(OpExpr, node);
+		Var *left_arg = linitial(op->args);
+		Var *right_arg = lsecond(op->args);
+
+		if (op->opno != context->eq_opr || !IsA(left_arg, Var) || !IsA(right_arg, Var))
+			return false;
+
+		/* Is this a join condition referencing our hypertable */
+		if ((left_arg->varno == context->sort_relid &&
```

Comment:
nit: this would be easier to read if the checks were in the same order for both sides of the OR. right now it's varno, varattno, varno vs varno, varno, varattno.